xbmc/Application.cpp

```diff
@@ -4211,9 +4213,9 @@ bool CApplication::OnMessage(CGUIMessage& message)
         CGUIMessage msg(GUI_MSG_PLAYLISTPLAYER_CHANGED, 0, 0, g_playlistPlayer.GetCurrentPlaylist(), param, item);
         g_windowManager.SendThreadMessage(msg);
         g_playlistPlayer.SetCurrentSong(m_nextPlaylistItem);
-        *m_itemCurrentFile = *item;
+        m_itemCurrentFile = item;
```

Comment:
item is not created in this block (since it's an element created as part of the playlist). The reason to deep copy was to not alter the original object in the playlist. There are some manipulations with the current item by CApplication (for playback, stacks, saving state, etc.) that you may not want to maintain, that's why there is deep copying.